### PR TITLE
AI Component Library: chat message component

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Creates a common "ChatMessage" component in the AI Component library. In this PR I've **only included it in AI Chat**, but will follow up with a change to integrate it into AI Tutor (since the PR was getting a bit big already). The existing "ChatNotificationMessages" in AI Chat can just be replaced by the DSCO Alert component directly, so the only new component is the chat message component.

Note: as with some other changes, using the DSCO Alert removes the existing formatting (bolding, spacing). I've opened a [task](https://codedotorg.atlassian.net/browse/LABS-902) to look into allowing Alert contents to be formatted and/or allowing the Alert component to render arbitrary content.

Screenshots:

**Messages with some code markdown**
Screenshot 2024-07-03 at 10 22 39 AM

**Messages with other markdown formatting**
Screenshot 2024-07-03 at 10 24 33 AM
Screenshot 2024-07-03 at 10 24 48 AM

**Alerts and other message conditions**
Screenshot 2024-07-03 at 10 31 58 AM

## Links

Figma: https://www.figma.com/design/tjkMqlAXi2iTqpUHSljB3A/Gen-AI-Unit?node-id=2653-6892&m=dev

## Testing story

Tested on a few Gen AI pilot levels

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @cearachew just in case this is useful for AI in Teacher Tools too

File: apps/src/aiComponentLibrary/chatItems/ChatMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  role: Role;
+  status: string;
+}
+
+const ChatMessage: React.FunctionComponent<ChatMessageProps> = ({
+  chatMessageText,
+  role,
+  status,
+}) => {
+  const hasDangerStyle =
```
Comment (Erin007) [inline apps/src/aiComponentLibrary/chatItems/ChatMessage.tsx:44]: "danger style" 😎

File: apps/src/aiComponentLibrary/chatItems/ChatMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const hasWarningStyle = status === Status.PII_VIOLATION;
+
+  return (
+    <div className={moduleStyles[`container-${role}`]}>
+      {role === Role.ASSISTANT && (
+        <div className={moduleStyles.botIconContainer}>
+          <img
+            src={aiBotIcon}
+            alt={commonI18n.aiChatBotIconAlt()}
```
Comment (Erin007) [inline apps/src/aiComponentLibrary/chatItems/ChatMessage.tsx:56]: 🎉

File: apps/src/aiComponentLibrary/chatItems/chat-message.module.scss [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  height: 32px;
+  background: $assistant-color;
+  border-radius: 20px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.botIcon {
+  width: 19.25px;
```
Comment (Erin007) [inline apps/src/aiComponentLibrary/chatItems/chat-message.module.scss:91]: TIL you can have decimals in pixel values

File: apps/src/aiComponentLibrary/chatItems/types.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/** Identifies the role of a sender for a given chat message. */
```
Comment (Erin007) [inline apps/src/aiComponentLibrary/chatItems/types.ts:1]: oh nice! I like the idea of sharing types where we can too

**Review by Erin007 — APPROVED** [review]
Looks great! Thanks, @sanchitmalhotra126 🎉

**Review by levadadenys — APPROVED** [review]
LGTM! Nice and clean! 😍

File: apps/src/aiComponentLibrary/chatItems/types.ts (hunk 4 above)
Comment (sanchitmalhotra126) [reply apps/src/aiComponentLibrary/chatItems/types.ts:1]: Yep! Figured this one at least made sense to share, but we can also find a better place to consolidate these if/when we share more types

**Pull request merged**